grails-app/services/org/pih/warehouse/requisition/RequisitionService.groovy

```diff
@@ -856,7 +856,8 @@ class RequisitionService {
                 requisition.rejectedBy = currentUser
                 break
             default:
-                break
+                requisition.status = newStatus
         }
+        requisition.save(flush: true)
```

Comment:
I think we should get rid of one of the saves - either this one or the one in the `transitionRequisitionStatus`.
If removing the `flush` won't cause any problems with the event that is published after this method is called, we could remove this save in my opinion.